### PR TITLE
1769/feat/flight num validation

### PR DESCRIPTION
**Pull request by y-golde**
# Flight Number Validation
## Background
This PR changes the two fields 'airline' and 'flight number', to be controlled by a list from the DB (similar to cities and street number), the fields are still being sent and stored as strings (at `public.exposure` that is).
_see ticket for more info_

### New Table `public.airlines` 
contains all of the available airlines and a unique id for each of them.
### New Table `public.flight_numbers`
contains all of the flight numbers available.
### `GET /airlines/`
gets all airlines <sup>Cached!</sup>
### `GET /airlines/flights/:airlineId`
gets all of the flight number by airline

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx [hunk 1]
```diff
@@ -93,7 +93,7 @@ const InvestigationInfoBar: React.FC<Props> = ({ currentTab }: Props) => {
             setEpidemiologyNum(lastOpenedEpidemiologyNumber);
             setLastOpenedEpidemiologyNum(defaultEpidemiologyNumber);
         } else {
-            handleInvalidEntrance();
+            process.env.REACT_APP_ENVIRONMENT !== 'local' && handleInvalidEntrance();
```
Comment (y-golde) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx:96]: [👍](https://media.giphy.com/media/H7TsKHN6xAoAy96fJr/giphy.gif)

Comment (Noam6421) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx:96]: Amazing!!!!!

File: client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/Forms/FlightsForm/FlightsForm.tsx [hunk 2]
```diff
@@ -12,15 +14,25 @@ import FlightNumberTextField from 'commons/FlightNumberTextField/FlightNumberTex
 
 import useStyles from './FlightFormStyles';
 import AirportInput from './AirportInput/AirportInput';
+import StoreStateType from 'redux/storeStateType';
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/Forms/FlightsForm/FlightsForm.tsx:17]: fix imports

File: client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/Forms/FlightsForm/FlightsForm.tsx [hunk 3]
```diff
@@ -45,9 +57,15 @@ const FlightsForm = (props: Props) => {
...
 
+	const getAirlineByDisplayName = (displayName : string) => {
+		return formattedAirlines.find(airline => airline.displayName === displayName);
+	};
+
     const currentErrors = errors ? (errors.exposures ? errors.exposures[index] : {}) : {};
     const startDateError = currentErrors ? currentErrors.flightStartDate : undefined;
     const endDateError = currentErrors ? currentErrors.flightEndDate : undefined;
+    const airlineError = currentErrors ? currentErrors.airline : undefined;
+	const flightNumError = currentErrors ? currentErrors[fieldsNames.flightNumber] : undefined;
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/Forms/FlightsForm/FlightsForm.tsx:80]: indentation

File: client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/Forms/FlightsForm/FlightsForm.tsx [hunk 4]
```diff
@@ -145,51 +163,71 @@ const FlightsForm = (props: Props) => {
...
-							}}
-						/>
+						<Grid item xs={3}>
+							<Controller
+								control={control}
+								name={`exposures[${index}].${fieldsNames.flightNumber}`}
+								defaultValue={exposureAndFlightsData[fieldsNames.flightNumber]}
+								render={(props) => {
+									return (
+											<Autocomplete
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/Forms/FlightsForm/FlightsForm.tsx:221]: move one tab left

File: client/src/components/App/Content/InvestigationForm/useInvestigationForm.ts [hunk 5]
```diff
@@ -29,6 +30,7 @@ import { setIdentificationTypes } from 'redux/IdentificationTypes/identification
 import UpdateTrackingRecommendation from 'Utils/TrackingRecommendation/updateTrackingRecommendation'; 
 
 import { useInvestigationFormOutcome } from './InvestigationFormInterfaces';
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/useInvestigationForm.ts:32]: fix imports order

File: client/src/components/App/Content/InvestigationForm/useInvestigationForm.ts [hunk 6]
```diff
@@ -78,6 +81,29 @@ const useInvestigationForm = (): useInvestigationFormOutcome => {
...
+            axios.get<Airline[]>('/airlines')
+                .then(result => {
+                    airlineLogger.info('request was successful', Severity.LOW);
+
+                    const airlinesMap = airlineListToMap(result.data);
+                    setAirlines(airlinesMap);
+                })
+                .catch(err => {
+                    airlineLogger.error(`recived error during request, err: ${err}`, Severity.HIGH);
+                    console.log(err);
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/useInvestigationForm.ts:98]: delete clg

File: server/src/ClientToDBAPI/AirlinesRoute/Airlines/getAirlines.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import { Request, Response } from 'express';
+import { Severity } from '../../../Models/Logger/types';
+import { ALL_AIRLINES } from '../../../DBService/Airlines/Query';
+import { errorStatusCode, graphqlRequest } from '../../../GraphqlHTTPRequest';
```
Comment (Noam6421) [inline server/src/ClientToDBAPI/AirlinesRoute/Airlines/getAirlines.ts:4]: fix imports order

File: server/src/ClientToDBAPI/mainRoute.ts [hunk 8]
```diff
@@ -18,6 +19,7 @@ import personalDetailsRoute from './PersonalDetailsRoute/mainRoute';
 import clinicalDetailsRoute from './ClinicalDetailsRoute/mainRoute';
 import groupedInvestigationsRoute from './GroupedInvestigationsRoute/mainRoute';
 
+
```
Comment (Noam6421) [inline server/src/ClientToDBAPI/mainRoute.ts:22]: delete

**Review by Noam6421 — CHANGES_REQUESTED** [review]

**Review by Noam6421 — APPROVED** [review]

**Review by chenFriedman — APPROVED** [review]

**Pull request merged**